BridgeAppSDK/SBAScheduledActivityManager.swift

```diff
@@ -278,7 +282,18 @@ public class SBAScheduledActivityManager: NSObject, SBASharedInfoController, ORK
             
             updateScheduledActivity(schedule, taskViewController: taskViewController)
             taskViewController.task?.updateTrackedDataStores(shouldCommit: true)
-            archiveResults(schedule, taskViewController: taskViewController)
+            guard let archives = archiveResults(schedule, taskViewController: taskViewController) else {
+                print("Archiving results failed")
+                return
+            }
+            for archive in archives {
+                archive.encryptAndUploadArchiveWithCompletion({ (error) in
+                    if error != nil {
+                        print("Error encrypting and uploading archive:\n\(error)")
+                    }
+                })
+            }
```

Comment:
Please put all this into archiveResults() as a void function. That way subclasses can override as needed and it is easier to refactor to allow customization that hasn't yet been thought of.